frontend/templates/product/sections/ProductOverviewSection/ProductGallery.tsx

```diff
@@ -248,75 +199,69 @@ function useCurrentImageIndex(
    return currentImageIndex;
 }
 
-type CustomNavigationType = {
-   swiper?: Swiper | null;
-   isBeginning?: boolean;
-   isEnd?: boolean;
+type NavigationButtonProps = {
+   disabled?: boolean;
+   onClick?: React.MouseEventHandler<HTMLButtonElement>;
 };
 
-const CustomNavigation = ({
-   swiper,
-   isBeginning,
-   isEnd,
-}: CustomNavigationType) => {
-   return (
-      <>
-         <Button
-            pos="absolute"
-            top="50%"
-            left="2"
-            transform="translateY(-50%)"
-            zIndex="1"
-            onClick={() => swiper?.slidePrev()}
-            disabled={isBeginning}
-            backgroundColor="transparent"
-            h="48px"
-            w="48px"
-            borderRadius="full"
-            role="group"
-            _hover={{
-               backgroundColor: 'transparent',
-            }}
-         >
-            <Circle
-               size="32px"
-               bg="gray.600"
-               _groupHover={!isBeginning ? { bg: 'gray.800' } : undefined}
-               transition="300ms all"
-            >
-               <FaIcon icon={faArrowLeft} color="white" />
-            </Circle>
-         </Button>
-         <Button
-            data-testid="swiper-next-image"
-            pos="absolute"
-            top="50%"
-            right="2"
-            transform="translateY(-50%)"
-            zIndex="1"
-            onClick={() => swiper?.slideNext()}
-            disabled={isEnd}
-            backgroundColor="transparent"
-            h="48px"
-            w="48px"
-            borderRadius="full"
-            role="group"
-            _hover={{
-               backgroundColor: 'transparent',
-            }}
-         >
-            <Circle
-               size="32px"
-               bg="gray.600"
-               _groupHover={!isEnd ? { bg: 'gray.800' } : undefined}
-               transition="300ms all"
-            >
-               <FaIcon icon={faArrowRight} color="white" />
-            </Circle>
-         </Button>
-      </>
-   );
-};
+const PreviousButton = ({ disabled, onClick }: NavigationButtonProps) => (
+   <Button
```

Comment:
Any particular reason these aren't separated into a shared component? Most props are repeated, except for left/right specifics.